src/cirrus/docs.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+"""
+_docs_
+
+Implement git cirrus docs command
+
+
+"""
+from argparse import ArgumentParser
+from cirrus.configuration import load_configuration
+from cirrus.documentation_utils import (
+    build_docs,
+    build_doc_artifact,
+    upload_documentation
+)
+
+
+def build_parser(argslist):
+    """
+    _build_parser_
+
+    Set up command line parser for the docs command
+
+    """
+    parser = ArgumentParser(
+        description='git cirrus docs command'
+    )
+    parser.add_argument('command', nargs='?')
+
+    subparsers = parser.add_subparsers(dest='command')
+    build_command = subparsers.add_parser(
+        'build',
+        help='Build Sphinx documentation by calling the make command'
+    )
+    build_command.add_argument(
+        '--make',
+        nargs='*',
+        help=('generate documentation with Sphinx (Makefile path must be set '
+              'in cirrus.conf) and create an artifact. Argument list is used '
+              'to run the Sphinx make command. Default: clean html')
+    )
+    build_command.set_defaults(make=[])
+
+    pack_command = subparsers.add_parser(
+        'pack',
+        help='Package documentation as a tarball'
+    )
+
+    upload_command = subparsers.add_parser(
+        'upload',
+        help='Upload documentation to a remote server'
+    )
+    upload_command.add_argument(
```

Comment:
I wonder if it wouldnt be cleaner to get these opts from a section of the cirrus config?  
Eg: for a package, you would decide docs go to a server, or docs get uploaded to readthedocs etc and set the plugin and options for that plugin in the cirrus conf?  
Seems like a cleaner workflow IMO